streamingvisitors/src/vespa/vsm/searcher/utf8stringfieldsearcherbase.h

```diff
@@ -53,14 +54,81 @@ class UTF8StringFieldSearcherBase : public StrChrFieldSearcher
         explicit OffsetWrapper(ucs4_t * buf, size_t * offsets) noexcept : BufferWrapper(buf), _boff(offsets), _coff(offsets) {}
         void onCharacter(ucs4_t ch, size_t of) { *_cbuf++ = ch; *_coff++ = of; }
         void onOffset(size_t of) { *_coff++ = of; }
-        bool valid() { return (size() == (size_t)(_coff - _boff)); }
-        bool hasOffsets() { return true; }
+        bool valid() const noexcept { return (size() == (size_t)(_coff - _boff)); }
+        bool hasOffsets() const noexcept { return true; }
     };
 
 protected:
     SharedSearcherBuf _buf;
 
-    const search::byte * tokenize(const search::byte * buf, size_t maxSz, cmptype_t * dstbuf, size_t & tokenlen);
+    using byte = search::byte;
+    using Normalizing = search::streaming::Normalizing;
+
+    class TokenizeReader {
```

Comment:
Perhaps this should be moved to a separate .h and .cpp file with its own basic unit test?

Reply:
Yes, I will do that in a separate PR.